src/bindings/jna/libelektra/src/main/java/org/libelektra/Elektra.java

```diff
@@ -0,0 +1,325 @@
+package org.libelektra;
+
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+
+import javax.annotation.Nullable;
+
+import org.libelektra.plugin.NativePlugin;
+
+/**
+ * The JNA interface to libelektra.
+ */
+public interface Elektra extends Library {
+
+	/**
+	 * Singleton instance of the native library proxy.
+	 */
+	Elektra INSTANCE = Native.loadLibrary("elektra-kdb", Elektra.class);
+
+	// KDB methods
+
+	/**
+	 * Opens the session with the Key database.<br >
```

Comment:
Why do you add such extensive documentation here. These functions are not supposed to be used directly, the interface is only a necessity to make JNA bindings work. Such a disclaimer should probably written next to "The JNA interface to libelektra.".

Reply:
I've also thought about that. I would argue that for JNA binding developers, these java docs are relevant.
The other side of the story is maintaining doc sync with the C API might be too much of a burden. The alternative would be to minimize the documentation here and link to the C API docs.

Reply:
I would prefer linking to it, as it is supposed to be a 1:1 representation of the C API.

Reply:
I will resort to minimal duplication still resulting in a helpful javadoc. will be addressed via #3754.